app/models/manageiq/consumption/showback_tier.rb

```diff
@@ -23,10 +23,67 @@ class ShowbackTier < ApplicationRecord
     validates :tier_start_value,  :numericality => {:greater_than_or_equal_to => 0, :less_than => Float::INFINITY}
     validates :tier_end_value,    :numericality => {:greater_than_or_equal_to => 0}
 
+    before_save :validate_interval
+
     def name
       "#{showback_rate.category}:#{showback_rate.measure}:#{showback_rate.dimension}:Tier:#{tier_start_value}-#{tier_end_value}"
     end
 
+    def validate_interval
+      ManageIQ::Consumption::ShowbackTier.where(:showback_rate => showback_rate).each do |tier|
+        next if tier == self
+        #returns true == overlap, false == no overlap
+        if tier.tier_end_value == Float::INFINITY and (tier_start_value>tier.tier_start_value or tier_end_value>tier.tier_start_value)
```

Comment:
It looks that in this case you need to modify + save all intervals at the same time to validate that we have everything covered.
Would not that be done better in the rate more than in the tiers?
http://blog.arkency.com/2014/05/mastering-rails-validations-objectify/